situp-plugins/service-map-stateful/src/main/java/com/amazon/situp/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -38,34 +38,30 @@
     private volatile static LmdbProcessorState<String> currentTraceGroupWindow;
     //TODO: Consider keeping this state in lmdb
     private volatile static  HashSet<ServiceMapRelationship> relationshipState = new HashSet<>();
-    private static File databasePath;
-    private static File traceDatabasePath;
+    private static File dbPath;
     private static Clock clock;
 
     private final int thisProcessorId;
 
     public ServiceMapStatefulProcessor(final PluginSetting pluginSetting) {
      this(pluginSetting.getIntegerOrDefault(ServiceMapProcessorConfig.WINDOW_DURATION, ServiceMapProcessorConfig.DEFAULT_WINDOW_DURATION)*TO_MILLIS,
-             new File(ServiceMapProcessorConfig.DEFAULT_SPANS_LMDB_PATH),
-             new File(ServiceMapProcessorConfig.DEFAULT_TRACES_LMDB_PATH),
+             new File(ServiceMapProcessorConfig.DEFAULT_LMDB_PATH),
              Clock.systemUTC());
     }
 
     ServiceMapStatefulProcessor(final long windowDurationMillis,
                                        final File databasePath,
-                                       final File traceDatabasePath,
                                        final Clock clock) {
         ServiceMapStatefulProcessor.clock = clock;
         this.thisProcessorId = processorsCreated.getAndIncrement();
         if(isMasterInstance()) {
             previousTimestamp = ServiceMapStatefulProcessor.clock.millis();
             ServiceMapStatefulProcessor.windowDurationMillis = windowDurationMillis;
-            ServiceMapStatefulProcessor.databasePath = createPath(databasePath);
-            ServiceMapStatefulProcessor.traceDatabasePath = createPath(traceDatabasePath);
-            currentWindow = new LmdbProcessorState<>(databasePath, getNewDbName(), ServiceMapStateData.class);
-            previousWindow = new LmdbProcessorState<>(databasePath, getNewDbName() + "-prev", ServiceMapStateData.class);
-            currentTraceGroupWindow = new LmdbProcessorState<>(traceDatabasePath, getNewTraceDbName(), String.class);
-            previousTraceGroupWindow = new LmdbProcessorState<>(traceDatabasePath, getNewTraceDbName() + "-prev", String.class);
+            ServiceMapStatefulProcessor.dbPath = createPath(databasePath);
+            currentWindow = new LmdbProcessorState<>(new File(dbPath.getPath() + getNewDbName()), "spansDb", ServiceMapStateData.class);
+            previousWindow = new LmdbProcessorState<>(new File(dbPath.getPath() + getNewDbName() + "-prev"), "spansDb", ServiceMapStateData.class);
+            currentTraceGroupWindow = new LmdbProcessorState<>(new File(dbPath.getPath() + getNewTraceDbName()), "traceDb", String.class);
+            previousTraceGroupWindow = new LmdbProcessorState<>(new File(dbPath.getPath() + getNewTraceDbName() + "-prev"), "traceDb", String.class);
```

Comment:
1. Why do we need `-prev`? When Situp starts for the first time there will be no previous right?
2. What is the logic for restart? If we are going with always start fresh policy? then wouldn't it mean we should have clean up job at the start up?

Reply:
1. It's a race condition that likely won't be hit, but also it's an issue that comes up in testing with a mock clock.
2. I think restart logic is not something that needs to be thought about and dealt with in another PR.

Reply:
Oh, I see what you mean. We can leave it null (but then we have to have null checks on the window)

Reply:
I think we shouldn't create empty files for the purpose of testing. I would prefer not to create previous window at all.

Reply:
Well its not for the purpose of testing. Its an empty file thats deleted after the first window rotation so that we don't have to have null checks every time we access either previous window just for the startup window.

Reply:
> Oh, I see what you mean. We can leave it null (but then we have to have null checks on the window)

I am okay with thanks checks.

Reply:
Having an empty file will make the code consistent, if it only 1 place where this is checked then null check is fine but if it is several than it is less error prone (now and in the future) to always have the same flow.  Having an empty DB for a period of minutes is not a big deal as no significant resources would be consumed.

Reply:
So its more than just a single null check. We need to null check when we want to iterate over the previous window, when we want to look for parent span, when we want to find trace group, when we want to rotate the windows, etc. Its 5 or 6 null checks, and some which would happen quite a lot.

Reply:
From  a complexity perspective I think we should keep this how you have it.
If this is a null check it means that that in order to have good test coverage we will need at most one new test per null check. Thats a lot more testing to avoid one empty DB file.

Reply:
I'm okay to have empty file, but can we at least remove the `-prev` suffix because its not a true suffix, we attach it only  during the first run, after which there is no `-prev`. Maybe `-empty`. 

(Again this logic will change when we introduce on restart, I am bit worried that we are changing the algo for every PR rather try to have a full fledged idea)

Reply:
Sure, I can change the suffix to "empty". I don't see how we are changing the algo every PR.